File: Samples/azure-storage/Azure.Java.Fluent/implementation/StorageAccountInner.java
/**
 */

package petstore.implementation;

import petstore.StorageAccountProperties;
import com.microsoft.azure.Resource;
import com.microsoft.azure.Resource;

/**
 * The storage account.
 */
public class StorageAccountInner extends Resource {
    /**
     * The properties property.
     */
    private StorageAccountProperties properties;

    /**
     * Get the properties value.
     *
     * @return the properties value
     */
    public StorageAccountProperties properties() {
        return this.properties;
    }

    /**
     * Set the properties value.
     *
     * @param properties the properties value to set
     * @return the StorageAccountInner object itself.
     */
    public StorageAccountInner withProperties(StorageAccountProperties properties) {
        this.properties = properties;
        return this;
    }

}

File: Samples/azure-storage/Azure.CSharp.Fluent/Models/StorageAccountInner.cs

namespace Petstore.Models
{
    using Microsoft.Rest;
    using Microsoft.Rest.Azure;
    using Newtonsoft.Json;
    using System.Collections;
    using System.Collections.Generic;
    using System.Linq;

    /// <summary>
    /// The storage account.
    /// </summary>
    public partial class StorageAccountInner : Microsoft.Rest.Azure.Resource
    {
        /// <summary>
        /// Initializes a new instance of the StorageAccountInner class.
        /// </summary>
        public StorageAccountInner()
        {
          CustomInit();
        }

        /// <summary>
        /// Initializes a new instance of the StorageAccountInner class.
        /// </summary>
        public StorageAccountInner(string location = default(string), string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), StorageAccountProperties properties = default(StorageAccountProperties))
            : base(location, id, name, type, tags)
        {
            Properties = properties;
            CustomInit();
        }

        /// <summary>
        /// An initialization method that performs custom operations like setting defaults
        /// </summary>
        partial void CustomInit();

        /// <summary>
        /// </summary>
        [JsonProperty(PropertyName = "properties")]
        public StorageAccountProperties Properties { get; set; }

        /// <summary>
        /// Validate the object.
        /// </summary>
        /// <exception cref="ValidationException">
        /// Thrown if validation fails
        /// </exception>
        public virtual void Validate()
        {
            if (Properties != null)
            {
                Properties.Validate();
            }
        }
    }
}
